applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c

```diff
@@ -165,26 +171,51 @@ static void base_recv_cb(struct bt_audio_broadcast_sink *sink, const struct bt_a
 {
 	int ret;
 	uint32_t base_bis_index_bitfield = 0U;
+	enum audio_channel channel;
 
 	if (init_routine_completed) {
 		return;
 	}
 
-	LOG_INF("Received BASE with %u subgroups from broadcast sink", base->subgroup_count);
+	/* Test to ensure there are enough streams for each subgroup */
+	if (base->subgroup_count >  CONFIG_BT_AUDIO_BROADCAST_SNK_STREAM_COUNT) {
+		LOG_INF("Too many channels in subgroup");
+		return;
+	}
+
+	/* TODO: Assign a headset mask for all channels of interest */
```

Comment:
I think there should be a Jira task to make the app number of channels agnostic.
Not sure it is loads of work and the data in the stream needs to carry a description (i.e. PTS, sample size, endianisum, interleaved or deinterleaved, etc) we would add a channel identifier to this for subsequent processing. Could offer a PCM formatting model that formats the data into the correct form on the way into the system via an ADC and on the way out to the DAC. This can be really useful if a vendor uses a different codec which generates or requires data in a particular form (e.g. LC3 encode uses deinterleaved buffers, SBC/Opus/MPEG/AAC all use and generate interleaved data).
And don't get me started on the sync module!

Reply:
Yeah, you can create several tasks, and then we can discuss them :)